src/app/user/modules/records/components/record-form/record-form.component.ts

```diff
@@ -71,6 +79,42 @@ export class RecordFormComponent implements OnInit {
       this.numHours.setValue(offCampusEvent.num_hours);
       this.numParticipants.setValue(offCampusEvent.num_participants);
     });
+    if (this.route.snapshot.queryParams.record_id !== undefined) {
+      const recordID = this.route.snapshot.queryParams.record_id;
+      this.recordService.getRecordWithDetail(recordID).subscribe(
+        record => {
+          this.toUpdate = true;
+          this.record = record;
+          this.record.off_campus_event = record.off_campus_event as OffCampusEvent;
```

Comment:
You have to make sure that `record.off_campus_event` is not null, malicious users may pass a invalid `record_id`.

Reply:
Furthermore, You might want to extract these assignment process in another function to make this function not so big.